File: 1-Iniciante/c/1047_tempo-de-jogo-com-minutos.c
#include <stdio.h>

int main () {
    int hi, mi, hf, mf, hora, minuto;
    scanf("%d %d %d %d", &hi, &mi, &hf, &mf);
    hora = hf - hi;
    minuto = mf -mi;
    if (hi == hf) {
        if (mi == mf) {
            hora = 24;
            minuto = 0;
        } else if (mi > mf) {
            hora = 24;
            hora -= 1;
            minuto += 60;
        } else {}
    } else if (hi > hf) {
        hora += 24;
        if (mi > mf) {
            minuto += 60;
            hora -= 1;
        } else {}
    } else {
        if (mi >= mf) {
            hora -= 1;
            minuto += 60;
        } else {}
    }

    printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", hora, minuto);

    return 0;
}

File: 1-Iniciante/python/1047_tempo-de-jogo-com-minutos.py
entrada = input().split()
hora_i, min_i, hora_f, min_f = list(map(int, entrada))
hora, minuto = hora_f - hora_i, min_f - min_i 
if hora_i == hora_f:     
    if min_i == min_f:
        hora = 24 
        minuto = 0 
    elif min_i > min_f:
        hora = 24
        hora -= 1
        minuto += 60
    else: pass   
elif hora_i > hora_f: 
    hora += 24
    if min_i > min_f:
        minuto += 60
        hora -= 1  
    else: pass
else: 
    if min_i >= min_f:
        hora -= 1
        minuto += 60
    else: pass

print(f'O JOGO DUROU {hora} HORA(S) E {minuto} MINUTO(S)')